x-pack/plugins/ml/public/application/jobs/jobs_list/components/datafeed_modal/datafeed_modal.tsx

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+import moment from 'moment';
+import {
+  EuiButtonEmpty,
+  EuiDatePicker,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLoadingChart,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalBody,
+  EuiSelect,
+  EuiSpacer,
+  EuiTabs,
+  EuiTab,
+} from '@elastic/eui';
+import {
+  Axis,
+  Chart,
+  CurveType,
+  LineSeries,
+  Position,
+  ScaleType,
+  Settings,
+  timeFormatter,
+} from '@elastic/charts';
+
+import { DATAFEED_STATE } from '../../../../../../common/constants/states';
+import { CombinedJobWithStats } from '../../../../../../common/types/anomaly_detection_jobs';
+import { useToastNotificationService } from '../../../../services/toast_notification_service';
+import { ml } from '../../../../services/ml_api_service';
```

Comment:
for new code it's better to use our api context rather than importing `ml` which uses the dependency cache

```
import { useMlApiContext } from '../../../../contexts/kibana';
```
```
const {
    results: { getDatafeedResultChartData },
  } = useMlApiContext();
```